pkg/operation/botanist/kubeapiserver.go

```diff
@@ -150,6 +156,20 @@ func (b *Botanist) computeKubeAPIServerAdmissionPlugins(defaultPlugins, configur
 	return admissionPlugins
 }
 
+func (b *Botanist) computeDisabledKubeAPIServerAdmissionPlugins(configuredPlugins []gardencorev1beta1.AdmissionPlugin, kubernetesVersion string) []gardencorev1beta1.AdmissionPlugin {
+	var disabledAdmissionPlugins []gardencorev1beta1.AdmissionPlugin
+	for _, plugin := range configuredPlugins {
+		if pointer.BoolDeref(plugin.Disabled, false) {
+			if versionutils.ConstraintK8sGreaterEqual125.Check(semver.MustParse(kubernetesVersion)) && plugin.Name == "PodSecurityPolicy" {
+				continue
+			}
```

Comment:
Instead of having this code here and silently ignoring information from the shoot spec, maybe we can rather make the `gardener-apiserver` remove this when the shoot is upgraded.

Concretely, in https://github.com/gardener/gardener/blob/c44dddf1439a0cbde65c7e9a8b54708b55656129/pkg/registry/core/shoot/strategy.go#L68-L78 we could check that if shoot is upgraded to 1.25 and PSP admission plugin is disabled then we can just drop the respective entry from the `Shoot`'s `.spec.kubernetes.kubeAPIServer.admissionPlugins`. This way, we don't leave "garbage" and can drop this client-side code here since everything will be handled by the server.

Reply:
Yes, me and @ialidzhikov discussed this, I will add handling for this.